pkg/util/slices.go

```diff
@@ -111,3 +111,11 @@ func InsertUnique[T cmp.Ordered](s []T, v T) []T {
 	}
 	return slices.Insert(s, idx, v)
 }
+
+// CopySlice creates a new slice and copies all elements from the provided
+// slice into it
+func CopySlice[T any, S []T](s S) S {
```

Comment:
consider calling this `Clone` to match the stdlib `slices.Clone`, or maybe `CloneSameSize` if you want to clarify the difference. This impl forces the slice to have the same exact size which is probably preferable here to avoid extra allocations.

Reply:
Ohh, i didn't know about slices.Clone. I created a benchmark to test the `CopySlice` function in this PR and slices.Clone and the bench diff showed there was no delta, so I've removed `CopySlice` and replace the usage with `slices.Clone`.